### PR TITLE
[flatbuffers] Update rust deps

### PR DESCRIPTION
**Pull request by ibc**
Hopefully it will fix https://github.com/versatica/mediasoup/pull/1064#issuecomment-1713617671

### CODE REVIEW

**Comment by ibc** [comment]
@nazar-pc while doing this I see this warning running `cargo clippy`:

```
$ cargo clippy
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
```

¯\_(ツ)_/¯

**Comment by ibc** [comment]
Wow, this is new...

https://github.com/versatica/mediasoup/actions/runs/6148272194/job/16681619576?pr=1155

```
    Checking mediasoup v0.11.4 (/home/runner/work/mediasoup/mediasoup/rust)
error: associated items `from_fbs` and `into_fbs` are never used
   --> rust/src/rtp_parameters.rs:707:19
    |
706 | impl RtpParameters {
    | ------------------ associated items in this implementation
707 |     pub(crate) fn from_fbs(
    |                   ^^^^^^^^
...
812 |     pub(crate) fn into_fbs(self) -> rtp_parameters::RtpParameters {
    |                   ^^^^^^^^
    |
    = note: `-D dead-code` implied by `-D warnings`

error: method `clock_rate` is never used
    --> rust/src/rtp_parameters.rs:1014:19
     |
994  | impl RtpCodecParameters {
     | ----------------------- method in this implementation
...
1014 |     pub(crate) fn clock_rate(&self) -> NonZeroU32 {
     |                   ^^^^^^^^^^
```

**Comment by nazar-pc** [comment]
> @nazar-pc while doing this I see this warning running cargo clippy:

Add `resolver = "2"` right after `[workspace]` in the root `Cargo.toml`

**Comment by ibc** [comment]
> Checking mediasoup v0.11.4 (/home/runner/work/mediasoup/mediasoup/rust)
> error: associated items `from_fbs` and `into_fbs` are never used

I'm fixing this by adding `#[allow(dead_code)]` in those unused items.

File: rust/src/rtp_parameters.rs [hunk 1]
```diff
@@ -704,6 +704,7 @@ pub struct RtpParameters {
 }
 
 impl RtpParameters {
+    #[allow(dead_code)]
```
Comment (nazar-pc) [inline rust/src/rtp_parameters.rs:709]: This is not a solution. Please don't add such things without TODO explaining why you have added it, why it is fine and when you expect to remove it.

Comment (ibc) [reply rust/src/rtp_parameters.rs:709]: Done:

```diff
diff --git a/rust/src/rtp_parameters.rs b/rust/src/rtp_parameters.rs
index 5ee7daea5..1eca5882e 100644
--- a/rust/src/rtp_parameters.rs
+++ b/rust/src/rtp_parameters.rs
@@ -704,6 +704,8 @@ pub struct RtpParameters {
 }
 
 impl RtpParameters {
+    /// TODO: This is not yet used but will be once flatbuffers is fully
+    /// implemented in mediasoup-rust.
     #[allow(dead_code)]
     pub(crate) fn from_fbs(
         rtp_parameters: rtp_parameters::RtpParameters,
@@ -1013,6 +1015,8 @@ impl RtpCodecParameters {
     }
 
     #[allow(dead_code)]
+    /// NOTE: This is not used in mediasoup code. However there is no reason to not
+    /// expose it as public API since it could be used by the user.
     pub(crate) fn clock_rate(&self) -> NonZeroU32 {
         let (Self::Audio { clock_rate, .. } | Self::Video { clock_rate, .. }) = self;
         *clock_rate
```

Comment (ibc) [reply rust/src/rtp_parameters.rs:709]: (typo fixed BTE, comment edited and new commit pushed)

Comment (ibc) [reply rust/src/rtp_parameters.rs:709]: ok, now imagine that I wrote `#[allow(dead_code)]` **below** the comment.

**Review by nazar-pc — APPROVED** [review]

**Comment by nazar-pc** [comment]
Too many notifications from these PRs, consider reviewing things before pushing, running Clippy locally, etc. And push all commits at once instead of one by one. With squashing you'll likely have to push just one because this PR already has 7 commits and a bunch of comments. This is very distracting.

**Comment by ibc** [comment]
> Too many notifications from these PRs, consider reviewing things before pushing, running Clippy locally, etc. And push all commits at once instead of one by one. With squashing you'll likely have to push just one because this PR already has 7 commits and a bunch of comments. This is very distracting.

I expected this to be a single commit PR and my computer is running other heavy things so `cargo clippy` never ended. The terrible mistakes and typos. Sorry.

**Pull request merged**